comp/forwarder/defaultforwarder/default_forwarder.go

```diff
@@ -502,6 +502,23 @@ func (f *DefaultForwarder) createAdvancedHTTPTransactions(endpoint transaction.E
 	for _, payload := range payloads {
 		for domain, dr := range f.domainResolvers {
 			drDomain, destinationType := dr.Resolve(endpoint) // drDomain is the domain with agent version if not local
+
+			if payload.Destination == transaction.PreaggrOnly {
+				preaggURL := f.config.GetString("preaggregation.dd_url")
+				primaryURL := f.config.GetString("dd_url")
```

Comment:
Primary url can be derived from `site` as well as specified explicitly, so the check below is probably going to miss common cases (see how GetMultipleEndpoints, that feeds domain resolver does it https://github.com/datadog/datadog-agent/blob/00a177679056c4d1e1edf913570bfbe8f63aa708/pkg/config/utils/endpoints.go#L142)